vespajlib/src/test/java/com/yahoo/geo/OneDegreeParserTestCase.java

```diff
@@ -0,0 +1,266 @@
+        checkZeroLat(false, "N0.0");
+        checkZeroLat(false, "N0\u00B00'0");
+        checkZeroLat(false, "S0");
+        checkZeroLat(false, "S0.0");
+        checkZeroLat(false, "S0o0'0");
+        checkZeroLat(false, "S0\u00B00'0");
+
+        checkZeroLon(true, "E0");
+        checkZeroLon(true, "E0.0");
+        checkZeroLon(true, "E0\u00B00'0");
+        checkZeroLon(true, "W0");
+        checkZeroLon(true, "W0.0");
+        checkZeroLon(true, "W0o0'0");
+        checkZeroLon(true, "W0\u00B00'0");
+    }
+
+    /**
+     * Tests inputs that are close to 0.
+     */
+    @Test
+    public void testNearZero() {
+        checkLat("N0.0001", 0.0001);
+        checkLat("S0.0001", -0.0001);
+        checkLon("E0.0001", 0.0001);
+        checkLon("W0.0001", -0.0001);
+
+        checkLat("N0.000001", 0.000001);
+        checkLat("S0.000001", -0.000001);
+        checkLon("E0.000001", 0.000001);
+        checkLon("W0.000001", -0.000001);
+
+        checkLat("N0\u00B00'1", 1/3600d);
+        checkLat("S0\u00B00'1", -1/3600d);
+        checkLon("E0\u00B00'1", 1/3600d);
+        checkLon("W0\u00B00'1", -1/3600d);
+    }
+
+    /**
+     * Tests inputs that are close to latitude 90/-90 degrees and longitude 180/-180 degrees.
+     */
+    @Test
+    public void testNearBoundary() {
+        checkLat("N89.9999", 89.9999);
+        checkLat("S89.9999", -89.9999);
+        checkLon("E179.9999", 179.9999);
+        checkLon("W179.9999", -179.9999);
+
+        checkLat("N89.999999", 89.999999);
+        checkLat("S89.999999", -89.999999);
+        checkLon("E179.999999", 179.999999);
+        checkLon("W179.999999", -179.999999);
+
+        checkLat("N89\u00B059'59", 89+59/60d+59/3600d);
+        checkLat("S89\u00B059'59", -(89+59/60d+59/3600d));
+        checkLon("E179\u00B059'59", 179+59/60d+59/3600d);
+        checkLon("W179\u00B059'59", -(179+59/60d+59/3600d));
+    }
+
+    /**
+     * Tests inputs that are on latitude 90/-90 degrees and longitude 180/-180 degrees.
+     */
+    @Test
+    public void testOnBoundary() {
+        checkLat("N90", 90d);
+        checkLat("N90\u00B00'0", 90d);
+        checkLat("S90", -90d);
+        checkLat("S90\u00B00'0", -90d);
+
+        checkLon("E180", 180d);
+        checkLon("E180\u00B00'0", 180d);
+        checkLon("W180", -180d);
+        checkLon("W180\u00B00'0", -180d);
+    }
+
+    /**
+     * Tests inputs that are above latitude 90/-90 degrees and longitude 180/-180 degrees.
+     */
+    @Test
+    public void testAboveBoundary() {
+        String message = "";
+        try {
+            parser = new OneDegreeParser(false, "N90.0001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
+        assertEquals("out of range [-90,+90]: 90.0001", message);
+        try {
+            parser = new OneDegreeParser(false, "S90.0001");
+        } catch (IllegalArgumentException e) {
+            message = e.getMessage();
+        }
```

Comment:
Consider a helper function, lines 148-153 is very similar to lines 154-159 and so on.